src/main/java/com/redhat/exhort/providers/GoModulesProvider.java

```diff
@@ -48,91 +47,49 @@
-    // Path.of("/home/zgrinber/git/exhort-java-api/src/test/resources/tst_manifests/golang/go_mod_light_no_ignore/go.mod");
-    Path path = Path.of("/tmp/xieshen/go.mod");
-    Provider provider = new GoModulesProvider();
-    GoModulesProvider goProvider = (GoModulesProvider) provider;
-    //    boolean answer = goProvider.IgnoredLine("        github.com/davecgh/go-spew v1.1.1 //
-    // indirect
-    // //exhortignore");
-    try {
-      //      provider.provideStack(path);
-      byte[] bytes = Files.readAllBytes(path);
-      provider.provideComponent(bytes);
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
-  }
-
-  public GoModulesProvider() {
-    super(Type.GOLANG);
+  public GoModulesProvider(Path manifest) {
+    super(Type.GOLANG, manifest);
     this.goEnvironmentVariableForPurl = getQualifiers(true);
-    this.goEnvironmentVariablesForRef = getQualifiers(false);
     this.mainModuleVersion = getDefaultMainModuleVersion();
   }
 
   @Override
-  public Content provideStack(final Path manifestPath) throws IOException {
+  public Content provideStack() throws IOException {
     // check for custom npm executable
-    Sbom sbom = getDependenciesSbom(manifestPath, true);
+    Sbom sbom = getDependenciesSbom(manifest, true);
     return new Content(
         sbom.getAsJsonString().getBytes(StandardCharsets.UTF_8), Api.CYCLONEDX_MEDIA_TYPE);
   }
 
   @Override
-  public Content provideComponent(byte[] manifestContent) throws IOException {
-    // check for custom npm executable
-    return new Content(
-        getDependenciesSbomCa(manifestContent).getAsJsonString().getBytes(StandardCharsets.UTF_8),
-        Api.CYCLONEDX_MEDIA_TYPE);
-  }
-
-  @Override
-  public Content provideComponent(Path manifestPath) throws IOException {
-    throw new IllegalArgumentException(
-        "provideComponent with file system path for GoModules package manager not implemented yet");
-  }
-
-  private Sbom getDependenciesSbomCa(byte[] manifestContent) {
-    Sbom sbom;
-    try {
-      Path tempRepository = Files.createTempDirectory("exhort-go");
-      Path path = Paths.get(tempRepository.toAbsolutePath().normalize().toString(), "go.mod");
-      Files.deleteIfExists(path);
-      Path manifestPath = Files.createFile(path);
-      Files.write(manifestPath, manifestContent);
-      sbom = getDependenciesSbom(manifestPath, false);
-
-      Files.delete(manifestPath);
-      Files.delete(tempRepository);
-    } catch (IOException e) {
-      throw new RuntimeException(e);
+  public Content provideComponent() throws IOException {
+    if (!Files.isRegularFile(manifest)) {
+      throw new IllegalArgumentException(
+          "The provided manifest is not a regular file: " + manifest);
     }
-    return sbom;
-  }
-
-  private PackageURL getRoot(String DependenciesGolang) {
-    return null;
+    if (!Files.exists(manifest)) {
+      throw new IllegalArgumentException("Missing required go.mod file: " + manifest.toString());
```

Comment:
`Files.exists` check should probably come before `Files.isRegularFile`. `Files.isRegularFile` will return false if the file doesnt exist, which means we would never surface "Missing required go.mod file" if the file doesnt exist

Reply:
nice catch, thanks